### ISSUE TITLE
ALTER FUNCTION .. SUPPORT fails with syntax error on worker

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#5844]
The following fails:
```sql
CREATE FUNCTION test_support_func(internal)
    RETURNS internal
    AS '/home/marco/postgresql-source/14/src/test/regress/regress.so', 'test_support_func'
    LANGUAGE C STRICT;

CREATE FUNCTION my_int_eq(int, int) RETURNS bool
  LANGUAGE internal STRICT IMMUTABLE PARALLEL SAFE
  AS $$int4eq$$;

-- With support function that knows it's int4eq, we get a different plan
ALTER FUNCTION my_int_eq(int, int) SUPPORT test_support_func;

NOTICE:  issuing ALTER FUNCTION public.my_int_eq(integer,integer);
DETAIL:  on server marco@localhost:1401 connectionId: 1
NOTICE:  issuing ROLLBACK
DETAIL:  on server marco@localhost:1402 connectionId: 2
ERROR:  syntax error at or near ";"
CONTEXT:  while executing command on localhost:1402
```

**Comment by onderkalaci** [comment]
fyi

> SUPPORT support_function
Set or change the planner support function to use for this function. See [Section 38.11](https://www.postgresql.org/docs/14/xfunc-optimization.html) for details. You must be superuser to use this option.

**Comment by marcocitus** [comment]
True, though I'm a bit concerned we're generating a garbage statement. Can that break add node? (esp. for community users who use superuser by default, or just stuff created by admin user)

**Comment by onderkalaci** [comment]
I guess we concluded that PG does not record dependencies for support functions: https://github.com/citusdata/citus/pull/5701#discussion_r805964596

Still, should be fixed, just sharing earlier discussion (where we missed this :/ )

**Comment by velioglu** [comment]
Fixed with #5849
